tests/php/StorefrontWooCommerceAdjacentProductsTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Tests for Storefront_WooCommerce_Adjacent_Products using WP_UnitTestCase.
+ *
+ * @package storefront
+ */
+
+class StorefrontWooCommerceAdjacentProductsTest extends WP_UnitTestCase {
+
+	/**
+	 * Next product should skip hidden products with identical publish dates.
+	 */
+	public function test_next_product_skips_hidden_with_same_date() {
+		$date         = '2024-01-01 10:00:00';
+		$current_post = $this->create_product( $date );
+		$hidden_post  = $this->create_product( $date, true );
```

Comment:
Nitpick: let's call it `hidden_next` similar to the next test where it's called `hidden_prev`

```suggestion
		$hidden_next  = $this->create_product( $date, true );
```